### PR TITLE
Template pptx

### PR DESCRIPTION
**Pull request by seb231**

### CODE REVIEW

File: src/kixi/plugsocket.clj [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+(defn create-powerpoint [{:keys [width height
+                                 slides]
+                          :or {width 1920
+                               height 1080}}]
+  (let [powerpoint (XMLSlideShow.)]
+    (.setPageSize powerpoint (Dimension. width height))
+    (run!
+     #(create-slide (:objects %) powerpoint)
+     (sort-by :slide-no slides))
```
Comment (otfrom) [inline src/kixi/plugsocket.clj]: why not just in the order of the passed in seq?

**Review by otfrom — APPROVED** [review]
just not sure about sorting the incoming seq of slides in the create-powerpoint function

File: src/kixi/plugsocket.clj (hunk 1 above)
Comment (seb231) [reply src/kixi/plugsocket.clj]: I was worried about losing the order, will the sequence stay ordered?

Comment (otfrom) [reply src/kixi/plugsocket.clj]: if it is a vector or a seq of maps being passed in (which I think it should be) then yes, the vector/list/seq will stay in order. maps and sets don't (usually) maintain order.

Comment (seb231) [reply src/kixi/plugsocket.clj]: Removed

**Review by otfrom — APPROVED** [review]
let's see where this will take us

File: src/kixi/plugsocket.clj [hunk 2]
```diff
@@ -1,25 +1,218 @@
...
+   :table-box table-box})
+
+(defn create-slide
+  ;; takes a sequence of maps corresponding to a number of objects
+  ;; (text boxes, tables, images) to display on a slide
+  ([powerpoint]
+   (.createSlide powerpoint))
+  ([seq-of-maps powerpoint]
+   (create-slide seq-of-maps slide-fns powerpoint))
+  ([seq-of-maps config powerpoint]
```
Comment (otfrom) [inline src/kixi/plugsocket.clj]: seq-of-maps could be called something like `slide-def` here, but I think this is fine.

**Review by otfrom — APPROVED** [review]
a possible name change for a parameter that doesn't affect the API

:shipit:

**Pull request merged**